main.tf

```diff
@@ -131,72 +168,54 @@ resource "google_sql_database" "database" {
   project  = var.project_id
   name     = "todo"
   instance = google_sql_database_instance.main.name
+  # I know, explicit dependencies.
+  # But if I didn't put that there, destroy doesn work
+  depends_on = [google_service_account.runsa]
 }
 
 
-resource "random_password" "password" {
-  length           = 16
-  special          = true
-  override_special = "!#$%&*()-_=+[]{}<>:?"
-}
-
 resource "google_sql_user" "main" {
   project  = var.project_id
-  name     = "todo_user"
-  password = random_password.password.result
+  name     = "${google_service_account.runsa.account_id}@${var.project_id}.iam"
+  type     = "CLOUD_IAM_SERVICE_ACCOUNT"
   instance = google_sql_database_instance.main.name
-}
+  # I know, explicit dependencies.
+  # But if I didn't put that there, destroy doesn work
+  depends_on = [google_service_account.runsa]
 
-# Looked at using the module, but there doesn't seem to be a huge win there.
-# Handle redis instance
-resource "google_redis_instance" "main" {
-  authorized_network      = module.network-safer-mysql-simple.network_name
-  connect_mode            = "DIRECT_PEERING"
-  location_id             = var.zone
-  memory_size_gb          = 1
-  name                    = "${var.deployment_name}-cache"
-  project                 = var.project_id
-  redis_version           = "REDIS_6_X"
-  region                  = var.region
-  reserved_ip_range       = "10.137.125.88/29"
-  tier                    = "BASIC"
-  transit_encryption_mode = "DISABLED"
-  depends_on              = [module.project-services]
-  labels                  = var.labels
 }
 
-
-
 module "secret-manager" {
   source     = "GoogleCloudPlatform/secret-manager/google"
   version    = "~> 0.1"
   project_id = var.project_id
   labels = {
-    redishost = var.labels,
-    sqlhost   = var.labels,
-    todo_user = var.labels,
-    todo_pass = var.labels
+    redis_host = var.labels,
+    db_host    = var.labels,
+    db_user    = var.labels,
+    db_conn    = var.labels,
```

Comment:
Since these are non secrets, should we remove secret manager all together and just use env vars?